src/Datadog.Trace.ClrProfiler.Managed/Integrations/HttpClientIntegration.cs

```diff
@@ -0,0 +1,146 @@
+
+            try
+            {
+                instrumentedMethod =
+                    MethodBuilder<Func<HttpClient, HttpRequestMessage, HttpCompletionOption, CancellationToken, Task<HttpResponseMessage>>>
+                       .Start(moduleVersionPtr, mdToken, opCode, SendAsync)
+                       .WithConcreteType(httpClient)
+                       .WithParameters(request, compOption, cancellationToken)
+                       .WithNamespaceAndNameFilters(ClrNames.GenericTask, ClrNames.HttpRequestMessage, ClrNames.HttpCompletionOption, ClrNames.CancellationToken)
+                       .Build();
+            }
+            catch (Exception ex)
+            {
+                Log.ErrorRetrievingMethod(
+                    exception: ex,
+                    moduleVersionPointer: moduleVersionPtr,
+                    mdToken: mdToken,
+                    opCode: opCode,
+                    instrumentedType: HttpClientTarget,
+                    methodName: SendAsync,
+                    instanceType: handler.GetType().AssemblyQualifiedName);
+                throw;
+            }
+
+            return SendAsyncInternal(
+                instrumentedMethod,
+                reportedType: callOpCode == OpCodeValue.Call ? httpClient : handler.GetType(),
+                (HttpClient)handler,
+                (HttpRequestMessage)request,
+                compOption,
+                cancellationToken);
+        }
+
+        private static async Task<HttpResponseMessage> SendAsyncInternal(
+            Func<HttpClient, HttpRequestMessage, HttpCompletionOption, CancellationToken, Task<HttpResponseMessage>> sendAsync,
+            Type reportedType,
+            HttpClient handler,
+            HttpRequestMessage request,
+            HttpCompletionOption completionOption,
+            CancellationToken cancellationToken)
+        {
+            if (!IsTracingEnabled(request))
+            {
+                // skip instrumentation
+                return await sendAsync(handler, request, completionOption, cancellationToken).ConfigureAwait(false);
+            }
+
+            string httpMethod = request.Method?.Method;
+
+            using (var scope = ScopeFactory.CreateOutboundHttpScope(Tracer.Instance, httpMethod, request.RequestUri, IntegrationName))
+            {
+                try
+                {
+                    HttpResponseMessage response = await sendAsync(handler, request, completionOption, cancellationToken).ConfigureAwait(false);
+
+                    // this tag can only be set after the response is returned
+                    scope?.Span.SetTag(Tags.HttpStatusCode, ((int)response.StatusCode).ToString());
+
+                    return response;
+                }
+                catch (Exception ex) when (scope?.Span.SetExceptionForFilter(ex) ?? false)
+                {
+                    // unreachable code
+                    throw;
+                }
```

Comment:
We have been moving away from using the exception filter in new integrations (or when we go back to clean up older ones). The code is harder to understand and provides no benefit. We should use this instead:
```suggestion
                catch (Exception ex)
                {
                    scope?.Span.SetException(ex);
                    throw;
                }
```

Reply:
I'll make the change and test.